### PR TITLE
22 crud pattern implementation post

### PR DESCRIPTION
**Pull request by mayurhyphenOs**

### CODE REVIEW

File: posts/crud-pattern-in-react-implementation.md [hunk 1]
```diff
@@ -1,217 +1,33 @@
...
 .. tags: ReactJS
+.. category: Mayur Borse
 .. link:
 .. description:
 .. type: text
-.. status: draft
-.. type: text
-.. summary: In this post, we will see the implementation details of <em>BookTable</em> and <em>BookForm</em> components. We will explore how to handle field change, field validation, form errors, API errors in <em>BookForm</em> component. We will be using an apiToFormFieldIDs object that holds validation properties for every field. We will also explore how to achieve <strong>Inverse Data Flow</strong> in <em>BookTable</em> by passing data back to parent component Books.
+.. status:
+.. summary: This is a continuation post of our <strong>CRUD Pattern</strong> series. We introduced the pattern with the <em>Books</em> (parent) component in the previous post <a href="/blog/crud-pattern-in-react-introduction">here</a>. This post will detail the implementation details of <em>BookForm</em> and <em>BookTable</em> (child) components.
```
Comment (gabhijit) [inline posts/crud-pattern-in-react-implementation.md:10]: Just add the link to `previous post` and then there is no need of `here`.

File: posts/crud-pattern-in-react-implementation.md [hunk 2]
```diff
@@ -1,217 +1,33 @@
...
 
+# Recap
```
Comment (gabhijit) [inline posts/crud-pattern-in-react-implementation.md:12]: Overview

File: posts/crud-pattern-in-react-implementation.md [hunk 3]
```diff
@@ -1,217 +1,33 @@
...
 
-React supports **Unidirectional/One-Way Data Flow** only. It means, the data is passed from parent to child as `props` which are immutable arguments and not the other way around. Our actions (update, delete) are defined in the `Books` component (parent), but we need the data (selected book id or object) from `BookTable` component (child) to perform the actions. It is not possible to pass data back in *Unidirectional Data Flow*. So how do we *pass back* data from child to parent component? We are able to achieve it by having an **Inverse Data Flow**.
+React supports **Unidirectional/One-Way Data Flow** only. It means, the data is passed from parent to child as `props` which are immutable arguments and not the other way around. Our actions (update, delete) are defined in the `Books` component (parent), but we need the data (selected book id or object) from `BookTable` component (child) to perform the actions. It is not possible to pass data back in _Unidirectional Data Flow_. So how do we _pass back_ data from child to parent component? We are able to achieve it by having an **Inverse Data Flow**.
```
Comment (gabhijit) [inline posts/crud-pattern-in-react-implementation.md:20]: No Bold / No Italic

Comment (gabhijit) [reply posts/crud-pattern-in-react-implementation.md:20]: React supports Unidirectional/One-Way Data Flow only. It means, the data is passed from parent to child as `props` which are immutable arguments ~~and not the other way around~~. Our actions (`update`, `delete`) are defined in the `Books` component (parent), but we need the data (selected book `id` or object) from `BookTable` component (child) to perform these actions. We can achieve this by explicitly mapping  _Inverse Data Flow_ from `BookTable` component to `Books` component.

File: posts/crud-pattern-in-react-implementation.md [hunk 4]
```diff
@@ -1,217 +1,33 @@
...
 
 ## Inverse Data Flow
 
 - A callback function defined in the parent component is passed as props to the child component.
 - The function is invoked by the child component for an event.
 - eg. to trigger update or delete action (function invoke) on button click (event).
```
Comment (gabhijit) [inline posts/crud-pattern-in-react-implementation.md:26]: Give the actual names of the actions

**Comment by mayurhyphenOs** [comment]
@gabhijit  review changes resolved

**Pull request closed without merging**